File: apps/patter-voice-sdk/libraries/python/getpatter/carriers/telnyx.py
"""Telnyx carrier credentials for Patter."""

from __future__ import annotations

import os
from dataclasses import dataclass

__all__ = ["Carrier"]


@dataclass(frozen=True)
class Carrier:
    """Telnyx carrier credentials.

    Phase 2 dispatch narrows on :attr:`kind` to instantiate the underlying
    :class:`getpatter.providers.telnyx_adapter.TelnyxAdapter` on demand.

    Example::

        from getpatter.carriers import telnyx

        carrier = telnyx.Carrier()                           # reads env
        carrier = telnyx.Carrier(api_key="...", connection_id="...")
    """

    api_key: str = ""
    connection_id: str = ""
    public_key: str = ""

    def __post_init__(self) -> None:
        key = self.api_key or os.environ.get("TELNYX_API_KEY", "")
        conn = self.connection_id or os.environ.get("TELNYX_CONNECTION_ID", "")
        pub = self.public_key or os.environ.get("TELNYX_PUBLIC_KEY", "")
        if not key or not conn:
            raise ValueError(
                "Telnyx carrier requires api_key and connection_id. Pass them "
                "explicitly or set TELNYX_API_KEY and TELNYX_CONNECTION_ID in "
                "the environment."
            )
        object.__setattr__(self, "api_key", key)
        object.__setattr__(self, "connection_id", conn)
        object.__setattr__(self, "public_key", pub)

    @property
    def kind(self) -> str:
        """Stable discriminator used for Phase 2 dispatch."""
        return "telnyx"

File: apps/patter-voice-sdk/libraries/typescript/src/telephony/telnyx.ts
/** Telnyx carrier credentials holder for Patter. */

/** Constructor options for the Telnyx {@link Carrier}. */
export interface TelnyxCarrierOptions {
  /** Telnyx API key. Falls back to TELNYX_API_KEY env var. */
  apiKey?: string;
  /** Telnyx connection ID. Falls back to TELNYX_CONNECTION_ID env var. */
  connectionId?: string;
  /** Optional Ed25519 public key for webhook signature verification. Falls back to TELNYX_PUBLIC_KEY env var. */
  publicKey?: string;
}

/**
 * Telnyx telephony carrier — holds API key, connection ID, and optional webhook public key.
 *
 * @example
 * ```ts
 * import * as telnyx from "getpatter/telephony/telnyx";
 * const carrier = new telnyx.Carrier();                     // reads env
 * const carrier = new telnyx.Carrier({ apiKey: "KEY...", connectionId: "123" });
 * ```
 */
export class Carrier {
  readonly kind = "telnyx" as const;
  readonly apiKey: string;
  readonly connectionId: string;
  readonly publicKey: string | undefined;

  constructor(opts: TelnyxCarrierOptions = {}) {
    const key = opts.apiKey ?? process.env.TELNYX_API_KEY;
    const conn = opts.connectionId ?? process.env.TELNYX_CONNECTION_ID;
    const pub = opts.publicKey ?? process.env.TELNYX_PUBLIC_KEY;
    if (!key) {
      throw new Error(
        "Telnyx carrier requires apiKey. Pass { apiKey: '...' } or " +
          "set TELNYX_API_KEY in the environment.",
      );
    }
    if (!conn) {
      throw new Error(
        "Telnyx carrier requires connectionId. Pass { connectionId: '...' } or " +
          "set TELNYX_CONNECTION_ID in the environment.",
      );
    }
    this.apiKey = key;
    this.connectionId = conn;
    this.publicKey = pub;
  }
}
